crates/sui-indexer/src/types_v2.rs

```diff
@@ -0,0 +1,473 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use move_core_types::language_storage::StructTag;
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use sui_json_rpc_types::{EndOfEpochInfo, ObjectChange};
+use sui_types::base_types::{ObjectDigest, SequenceNumber};
+use sui_types::base_types::{ObjectID, SuiAddress};
+use sui_types::digests::TransactionDigest;
+use sui_types::dynamic_field::DynamicFieldInfo;
+use sui_types::effects::TransactionEffects;
+use sui_types::messages_checkpoint::{CheckpointDigest, EndOfEpochData};
+use sui_types::move_package::MovePackage;
+use sui_types::object::{Object, Owner};
+use sui_types::sui_serde::SuiStructTag;
+use sui_types::sui_system_state::sui_system_state_summary::SuiValidatorSummary;
+use sui_types::transaction::SenderSignedData;
+
+use crate::errors::IndexerError;
+
+pub type IndexerResult<T> = Result<T, IndexerError>;
+
+#[derive(Debug)]
+pub struct IndexedCheckpoint {
+    pub sequence_number: u64,
+    pub checkpoint_digest: CheckpointDigest,
+    pub epoch: u64,
+    pub tx_digests: Vec<TransactionDigest>,
+    pub network_total_transactions: u64,
+    pub previous_checkpoint_digest: Option<CheckpointDigest>,
+    pub end_of_epoch: bool,
+    pub timestamp_ms: u64,
+    pub total_gas_cost: i64, // total gas cost could be negative
+    pub computation_cost: u64,
+    pub storage_cost: u64,
+    pub storage_rebate: u64,
+    pub non_refundable_storage_fee: u64,
+    pub checkpoint_commitments: Vec<u8>,
+    pub validator_signature: Vec<u8>,
+    pub successful_tx_num: usize,
```

Comment:
successful_tx_num is indexed but not persisted in the DB, why is that?

Reply:
The plan is to this to calculate real time tps, and that will be streaming `successful_tx_num` per checkpoint without reading it from db